Com.OneSignal/OneSignal.cs

```diff
@@ -4,43 +4,39 @@
 
 namespace Com.OneSignal
 {
-    /// <summary>
-    /// Cross platform MediaManager implemenations
-    /// </summary>
-    public class OneSignal
-    {
-        static readonly Lazy<IOneSignal> Implementation = new Lazy<IOneSignal>(CreateOneSignal);
+   public class OneSignal
+   {
+      public const string kOSSettingsKeyAutoPrompt = "kOSSettingsKeyAutoPrompt";
+      public const string kOSSettingsKeyInAppLaunchURL = "kOSSettingsKeyInAppLaunchURL";
 
-        /// <summary>
-        /// Current settings to use
-        /// </summary>
-        public static IOneSignal Current
-        {
-            get
-            {
+      static readonly Lazy<OneSignalShared> Implementation = new Lazy<OneSignalShared>(CreateOneSignal);
 
-                if (Implementation.Value == null)
-                {
-                    throw NotImplementedInReferenceAssembly();
-                }
-                return Implementation.Value;
+      public static OneSignalShared Current
+      {
+         get
+         {
+            if (Implementation.Value == null)
+            {
+               throw NotImplementedInReferenceAssembly();
             }
-        }
+            return Implementation.Value;
+         }
+      }
 
-        static IOneSignal CreateOneSignal()
-        {
+      static OneSignalShared CreateOneSignal()
```

Comment:
Use the name `OneSignalShared` instead of `IOneSignal` is really confusing and against C# guidelines. Would it be possible to keep this to IOneSignal?